src/librustc_codegen_ssa/back/link.rs

```diff
@@ -125,10 +119,15 @@ fn link_binary_output<'a, B: ArchiveBuilder<'a>>(sess: &'a Session,
             .tempdir_in(out_filename.parent().unwrap())
             .unwrap_or_else(|err| sess.fatal(&format!("couldn't create a temp dir: {}", err)));
         let metadata = emit_metadata(sess, codegen_results, &metadata_tmpdir);
-        if let Err(e) = fs::rename(metadata, &out_filename) {
-            sess.fatal(&format!("failed to write {}: {}", out_filename.display(), e));
+        match fs::rename(&metadata, &out_filename) {
+            Ok(_) => {
+                if sess.opts.debugging_opts.emit_directives {
+                    sess.parse_sess.span_diagnostic.maybe_emit_json_directive(
+                        format!("metadata file written: {}", out_filename.display()));
```

Comment:
Was my suggestion about emitting this on *any* output, not just metadata, and letting Cargo pick, discussed at all?
It seems possible that GitHub may have hid it so nobody say it...

Reply:
It's a reasonable idea, if there is a clear use in mind. But it would have been massive scope-creep for this PR, which is about doing the minimum for experimenting with pipelining.